### PR TITLE
Change config panel navigation

### PR DESCRIPTION
**Pull request by bramkragten**
The way we used the available space in the config panel was a bit useless at a lot of places so I thought we could use if for better things.

When we have enough space we now show the config navigation on the left.

It can use more fine tuning, this is a start.

image

### CODE REVIEW

**Comment by frenck** [comment]
❤️ Love it!

Maybe not for now, but we could consider adding icons to each item in the list, which makes is visually recognizable.

**Comment by bramkragten** [comment]
Yeah that would be nice, have to make sure there stays enough room for the content though, especially the data tables need the horizontal space.

**Comment by balloob** [comment]
Can we make the menu component be shown underneath the toolbar of the page? So automations is above the menu. And we highlight the current section?

**Comment by balloob** [comment]
Or, we drop the toolbar sharing between menu and content completely. Make the page really stand on their own

**Comment by balloob** [comment]
Like, start with dropping the automations toolbar first. Since there is a title underneath, should look ok?

**Comment by bramkragten** [comment]
Like this?
 
image

(btw `Automatiseringsbewerker` really?)

**Comment by bramkragten** [comment]
Not a fan of dropping the toolbar, it gets messy with subpages, etc.

**Comment by iantrich** [comment]
Not sure how I feel about this. The sidebar | config nav seem to conflict with each other

**Comment by bramkragten** [comment]
Depends how wide your screen is right now, if we have the space we do

**Comment by balloob** [comment]
Something like that, but drop the blue and align the word "Configuration" with the text in the menu.  No need to draw attention to something that's not important.

**Comment by SeanPM5** [comment]
Here's another potential idea / mockup tweaked from the screenshot in OP. This keeps the toolbar intact but tweaks the left config navigation a little bit. 

Changed the background color to match with the hamburger menu so it no longer sticks out, centered the "Configuration" text, and added a subtle breadcrumb icon (`mdi:chevron-right-circle`) to indicate the hierarchy.  Also made the active panel have bold text.

image

**Comment by bramkragten** [comment]
Not sure about that arrow, feels like something I could click on?

**Comment by bramkragten** [comment]
image

**Comment by iantrich** [comment]
I would ditch the blue and just have a header like you do in the list of possible configs like you did [here](https://github.com/home-assistant/home-assistant-polymer/pull/4377#issuecomment-572627039) and align it left

**Comment by SeanPM5** [comment]
> Not sure about that arrow, feels like something I could click on?

Yeah, agreed. Was going for a breadcrumbs-style look and thought it'd be kinda neat if the design reflected how people actually talk about it (the docs for example saying "Go to Configuration > Automation" etc). But you're right, that arrow would probably just confuse people into thinking it's a clickable button.

I do like your latest revision. 

I'm not convinced the blue toolbar should be dropped, it's there for literally every other section. I think it'd feel a bit weird and inconsistent if it's gone from just this one place? Makes it feel kinda like a bug or that the page didn't load properly IMO. Also people like to theme their HA and now it wouldn't be using their colors if they had a larger width? Not sure about that.

**Comment by iantrich** [comment]
> I'm not convinced the blue toolbar should be dropped, it's there for literally every other section. I think it'd feel a bit weird and inconsistent if it's gone from just this one place? Makes it feel kinda like a bug or that the page didn't load properly IMO. Also people like to theme their HA and now it wouldn't be using their colors if they had a larger width? Not sure about that.

yeah, i think you're right

File: src/layouts/hass-subpage.ts [hunk 1]
```diff
@@ -9,12 +9,16 @@ import {
...
 import "../components/ha-paper-icon-button-arrow-prev";
+import { classMap } from "lit-html/directives/class-map";
 
 @customElement("hass-subpage")
 class HassSubpage extends LitElement {
   @property()
   public header?: string;
-
+  @property({ type: Boolean })
+  public back = true;
```
Comment (balloob) [inline src/layouts/hass-subpage.ts]: Can you add a description ? What does it mean for `back` to be true ?

Comment (balloob) [reply src/layouts/hass-subpage.ts]: Or is this `showBackButton` ?

File: src/layouts/hass-subpage.ts [hunk 2]
```diff
@@ -9,12 +9,16 @@ import {
...
+  @property({ type: Boolean })
+  public toolbar = true;
```
Comment (balloob) [inline src/layouts/hass-subpage.ts]: `showToolbar` ?

File: src/panels/config/cloud/account/cloud-account.js [hunk 3]
```diff
@@ -63,7 +63,10 @@ class CloudAccount extends EventsMixin(LocalizeMixin(PolymerElement)) {
           color: var(--primary-color);
         }
       </style>
-      <hass-subpage header="[[localize('ui.panel.config.cloud.caption')]]">
+      <hass-subpage
+        back="[[!isWide]]"
```
Comment (balloob) [inline src/panels/config/cloud/account/cloud-account.js]:
```suggestion
        .back="[[!isWide]]"
```

File: src/panels/config/cloud/account/cloud-account.js [hunk 4]
```diff
@@ -63,7 +63,10 @@ class CloudAccount extends EventsMixin(LocalizeMixin(PolymerElement)) {
...
+        header="[[localize('ui.panel.config.cloud.caption')]]"
```
Comment (balloob) [inline src/panels/config/cloud/account/cloud-account.js]:
```suggestion
        .header="[[localize('ui.panel.config.cloud.caption')]]"
```

File: src/panels/config/cloud/login/cloud-login.js [hunk 5]
```diff
@@ -72,7 +72,10 @@ class CloudLogin extends LocalizeMixin(
           color: var(--secondary-text-color);
         }
       </style>
-      <hass-subpage header="[[localize('ui.panel.config.cloud.caption')]]">
+      <hass-subpage
+        back="[[!isWide]]"
```
Comment (balloob) [inline src/panels/config/cloud/login/cloud-login.js]: props

File: src/panels/config/core/ha-config-core.js [hunk 6]
```diff
@@ -33,7 +33,10 @@ class HaConfigCore extends LocalizeMixin(PolymerElement) {
         }
       </style>
 
-      <hass-subpage header="[[localize('ui.panel.config.core.caption')]]">
+      <hass-subpage
+        header="[[localize('ui.panel.config.core.caption')]]"
```
Comment (balloob) [inline src/panels/config/core/ha-config-core.js]: props

File: src/panels/config/ha-config-router.ts [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+          ),
+      },
+      dashboard: {
+        tag: "ha-config-dashboard",
+        load: () =>
+          import(
+            /* webpackChunkName: "panel-config-dashboard" */ "./dashboard/ha-config-dashboard"
+          ),
+      },
+      entity_registry: {
```
Comment (balloob) [inline src/panels/config/ha-config-router.ts]: Can we make this `entitites` ?

File: src/panels/config/person/ha-config-person.ts [hunk 8]
```diff
@@ -48,7 +48,10 @@ class HaConfigPerson extends LitElement {
     }
     const hass = this.hass;
     return html`
-      <hass-subpage header=${hass.localize("ui.panel.config.person.caption")}>
+      <hass-subpage
+        header=${hass.localize("ui.panel.config.person.caption")}
```
Comment (balloob) [inline src/panels/config/person/ha-config-person.ts]:
```suggestion
        .header=${hass.localize("ui.panel.config.person.caption")}
```

File: src/panels/config/scene/ha-scene-dashboard.ts [hunk 9]
```diff
@@ -30,14 +30,17 @@ import { forwardHaptic } from "../../../data/haptics";
 class HaSceneDashboard extends LitElement {
   @property() public hass!: HomeAssistant;
   @property() public narrow!: boolean;
+  @property() public isWide!: boolean;
   @property() public scenes!: SceneEntity[];
 
   protected render(): TemplateResult | void {
     return html`
       <hass-subpage
+        style="width: 100%"
```
Comment (balloob) [inline src/panels/config/scene/ha-scene-dashboard.ts]: Isn't this the default ?

**Review by balloob — APPROVED** [review]

File: src/panels/config/cloud/account/cloud-account.js (hunk 3 above)
Comment (bramkragten) [reply src/panels/config/cloud/account/cloud-account.js]: Polymer

File: src/panels/config/cloud/account/cloud-account.js (hunk 4 above)
Comment (bramkragten) [reply src/panels/config/cloud/account/cloud-account.js]: Polymer

File: src/panels/config/cloud/login/cloud-login.js (hunk 5 above)
Comment (bramkragten) [reply src/panels/config/cloud/login/cloud-login.js]: Polymer

File: src/panels/config/core/ha-config-core.js (hunk 6 above)
Comment (bramkragten) [reply src/panels/config/core/ha-config-core.js]: Polymer

**Comment by bramkragten** [comment]
Does any one know if this will work?
image

**Comment by balloob** [comment]
Add the whole prefix to both the search and replace values.

**Pull request merged**